src/main/java/com/minecolonies/coremod/entity/ai/citizen/miner/Node.java

```diff
@@ -273,14 +281,74 @@ public Vec2i getWestNodeCenter()
         return new Vec2i(getX() - DISTANCE_TO_NEXT_NODE, getZ());
     }
 
+    /**
+     * Return a random next node to work at, might be at this node or at a parent.
+     * @return the next node to go to.
+     * @param level the level it is part of.
+     */
+    @Nullable
+    public Node getRandomNextNode(final Level level, final int step)
+    {
+        if (step > 3)
+        {
+            return null;
+        }
+
+        final Node nextNode;
+        switch (random.nextInt(3))
+        {
+            case 0:
+                nextNode = level.getOpenNode(getNorthNodeCenter());
+                break;
+            case 1:
+                nextNode = level.getOpenNode(getSouthNodeCenter());
+                break;
+            case 2:
+                nextNode = level.getOpenNode(getEastNodeCenter());
+                break;
+            default:
+                nextNode = level.getOpenNode(getWestNodeCenter());
+        }
+
+        if (nextNode == null || nextNode.style == NodeType.SHAFT)
+        {
+            final Node parent = level.getOpenNode(getParent());
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Rename "parent" which hides the field declared at line 55. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AHiddenFieldCheck)